stickyuploads/storage.py

```diff
@@ -12,4 +15,18 @@ def __init__(self):
 
     def url(self, name):
         """Files are not accessable via url."""
-        raise NotImplementedError()
\ No newline at end of file
+        raise NotImplementedError()
+
+    def get_available_name(self, name, max_length=None):
+        """Return path to name placed in two random directories"""
+        name = os.path.join(
```

Comment:
[`tempfile.mkdtemp`][0] to generate the random directory. I'm not sure why this needs two directories.

[0]: https://docs.python.org/3/library/tempfile.html#tempfile.mkdtemp